src/mouse_events.cpp

```diff
@@ -597,6 +597,21 @@ void mouse_handler::mouse_motion(int x, int y, const bool browse, bool update, m
 	}
 }
 
+// Hook for notifying lua game kernel of mouse button events. We pass button as
+// a serpaate argument than the original SDL event in order to manage touch
+// emulation (i.e., long touch = right click) and such.
+void mouse_handler::mouse_button_event(const SDL_MouseButtonEvent& event, uint8_t button, map_location loc)
+{
+	if (loc == map_location::null_location())
+		return;
+
+	if(game_lua_kernel* lk = pc_.gamestate().lua_kernel_.get()) {
+		// Ever any reason to allow lua functions to consume an event? If so,
+		// would have to make sure state stays correctly updated.
```

Comment:
I think the answer here is yes, iirc one of the main complaints of the existing lua mouse clicked event was that there is no option to consume the event.

Reply:
ah hah! Well then let's doing right. I'm not 100% versed in every mousey use-case in Wesnoth, as we can be in a drag event at the mouse up. I think I need to study the code a bit more to be clear on how far we drag before we start considering it a drag event and such.

Maybe it would be a good idea to separate it out into mouse up, down and click with the later only occurring if the up event is in the same hex as the down -- I can be a sloppy clicker sometimes, but this should be a big enough target! For the sake of the lua support, I'm only supporting mouse button events on hex tiles, so that would probably be a reasonable way to distinguish a mouse up that's part of a drag event and one that should be considered a real "click"